tests/components/smtp/test_notify.py

```diff
@@ -139,3 +68,100 @@ async def test_reload_notify(hass):
 
 def _get_fixtures_base_path():
     return path.dirname(path.dirname(path.dirname(__file__)))
+
+
+@pytest.fixture
+def message():
+    """Return MockSMTP object with test data."""
+    mailer = MockSMTP(
+        "localhost",
+        25,
+        5,
+        "test@test.com",
+        1,
+        "testuser",
+        "testpass",
+        ["recip1@example.com", "testrecip@test.com"],
+        "Home Assistant",
+        0,
+    )
+    yield mailer
+
+
+HTML = """
+        <!DOCTYPE html>
+        <html lang="en" xmlns="http://www.w3.org/1999/xhtml">
+            <head><meta charset="UTF-8"></head>
+            <body>
+              <div>
+                <h1>Intruder alert at apartment!!</h1>
+              </div>
+              <div>
+                <img alt="tests/testing_config/notify/test.jpg" src="cid:tests/testing_config/notify/test.jpg"/>
+              </div>
+            </body>
+        </html>"""
+
+
+EMAIL_DATA = [
+    (
+        "Test msg",
+        {"images": ["tests/testing_config/notify/test.jpg"]},
+        "Content-Type: multipart/related",
+    ),
+    (
+        "Test msg",
+        {"html": HTML, "images": ["tests/testing_config/notify/test.jpg"]},
+        "Content-Type: multipart/related",
+    ),
+    (
+        "Test msg",
+        {"html": HTML, "images": ["test.jpg"]},
+        "Content-Type: multipart/related",
+    ),
+    (
+        "Test msg",
+        {"html": HTML, "images": ["tests/testing_config/notify/test.pdf"]},
+        "Content-Type: multipart/related",
+    ),
+]
+
+
+@pytest.mark.parametrize(
+    "message_data, data, content_type",
+    EMAIL_DATA,
+    ids=[
+        "Tests when sending text message and images.",
+        "Tests when sending text message, HTML Template and images.",
+        "Tests when image does not exist at mentioned location.",
+        "Tests when image type cannot be detected or is of wrong type.",
+    ],
+)
+def test_send_message(message_data, data, content_type, hass, message):
+    """Verify if we can send messages of all types correctly."""
+    sample_email = "<mock@mock>"
+    with patch("email.utils.make_msgid", return_value=sample_email):
+        result = message.send_message(message_data, data=data)
+        assert content_type in result
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6333224)